src/gfx_d3d12.zig

```diff
@@ -580,25 +584,33 @@ pub const D3D12State = struct {
         var new_mesh = Mesh{
             .vertex_buffer = vertex_buffer,
             .index_buffer = index_buffer,
-            .num_lods = mesh.num_lods,
-            .lods = undefined,
-            .bounding_box = .{
-                .min = [3]f32{ mesh.bounding_box.min[0], mesh.bounding_box.min[1], mesh.bounding_box.min[2] },
-                .max = [3]f32{ mesh.bounding_box.max[0], mesh.bounding_box.max[1], mesh.bounding_box.max[2] },
-            },
+            .sub_mesh_count = mesh.sub_mesh_count,
+            .sub_meshes = undefined,
+            .bounding_box = undefined,
         };
 
-        // 1. Update mesh's lods vertex and index offsets
-        {
-            var i: u32 = 0;
-            while (i < mesh.num_lods) : (i += 1) {
-                new_mesh.lods[i].vertex_offset = mesh.lods[i].vertex_offset;
-                new_mesh.lods[i].index_offset = mesh.lods[i].index_offset;
-                new_mesh.lods[i].vertex_count = mesh.lods[i].vertex_count;
-                new_mesh.lods[i].index_count = mesh.lods[i].index_count;
+        // 1. Update sub meshes' lods vertex and index offsets
+        for (0..mesh.sub_mesh_count) |sub_mesh_index| {
+            new_mesh.sub_meshes[sub_mesh_index] = SubMesh{
+                .lod_count = mesh.sub_meshes[sub_mesh_index].lod_count,
+                .lods = undefined,
+                .bounding_box = .{
+                    .min = [3]f32{ mesh.sub_meshes[sub_mesh_index].bounding_box.min[0], mesh.sub_meshes[sub_mesh_index].bounding_box.min[1], mesh.sub_meshes[sub_mesh_index].bounding_box.min[2] },
```

Comment:
I would extract `mesh.sub_meshes[sub_mesh_index].bounding_box` into a variable first to keep the line length down and also make it clear that they are all looking at the same bbox.